Component/delete button

PReichetanz opened the pull request:
Add DeleteButton as controlled component
- Controlled parts: content and deleteFunction
- Add style to button
- Add two stories ("X" and "–")

<img width="50" alt="Bildschirmfoto 2021-11-24 um 09 42 42" src="https://user-images.githubusercontent.com/33567019/143204144-0e3b5045-2b26-4f78-8bc1-3754c92f0ca9.png">
<img width="50" alt="Bildschirmfoto 2021-11-24 um 09 43 07" src="https://user-images.githubusercontent.com/33567019/143204149-fec61325-916b-42de-aebb-6069d4075348.png">

Closes #7

PReichetanz committed:
Add tertiary hover color for DeleteButton

PReichetanz committed:
Add DeleteButton as controlled component

PReichetanz committed:
Add two stories for DeleteButton

Story 1: content "X"
Story 2: content "–"

PReichetanz committed:
Refactor DeleteButton

Add container to center content and make perfect cirle

marwinburesch commented on src/app/components/DeleteButton/DeleteButton.tsx:
```diff
@@ -0,0 +1,41 @@
+
+type DeleteButtonProps = {
+  handleDelete: () => void;
+  content: string;
```
Maybe you want to use `children` here, that way you could add an SVG to display on the button in the future

PReichetanz replied:
Very good idea, thank you!

marwinburesch commented on src/app/components/DeleteButton/DeleteButton.tsx:
```diff
@@ -0,0 +1,41 @@
+  }
+`;
+
+const Delete = styled.button`
```
Not the best name for a styled button in my opinion

PReichetanz replied:
True, I changed it.

marwinburesch reviewed (commented):
Nice!
I have a little food for thought here though: If you'll ever need a similar looking button, you could turn this into a `RoundActionButton`, or in other words make this a tad more generic and you'll have a versatile round button :)

PReichetanz committed:
Update according to review

- Change name of DeleteButton to RoundActionButton
- Change "content" prop to children
- Rename styled componend "Delete" to "Button"
- Update stories

lajanzen commented on src/app/components/DeleteButton/DeleteButton.tsx:
```diff
@@ -0,0 +1,41 @@
+  border-radius: 50%;
+  border: 1px solid var(--color-stroke);
+  padding: 0.625rem;
+  position: absolute;
```
Not sure if I would make the whole component absolute. I'd rather leave this out here and when you integrate the component on the page, set the value there.

lajanzen reviewed (commented):
I think that the way you constructed the button with a container, the only "clickable" part of the button is actually the "X" or "-". 
In case you want the whole container to be clickable, I would suggest to remove the container and just style the whole button as such, so the whole component is clickable.
